src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ZERO_INDEX = 0;
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE
+            = "Index %d out of bounds for length %d";
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            array = resizeArr(array);
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == array.length) {
+            array = resizeArr(array);
+        }
+        if (index < size && index >= ZERO_INDEX) {
+            array = addingElementInside(value, index, array);
+        } else if (index == size) {
+            add(value);
+            return;
+        } else {
```

Comment:
try to simplify this if-else construction

Reply:
we can have have one if to check if the index is out of bounds to throw exception. and one if-else construction: if index == size - add to the end; else - line 27 to 32. **AND** think about using only one `System.arraycopy()`, you don't need two of them